lib/Matrix/models/messageeventmodel.cpp

```diff
@@ -162,117 +329,152 @@ QVariant MessageEventModel::data(const QModelIndex& index, int role) const
                 }
+                case MessageEventType::File:
+                case MessageEventType::Audio:
+                case MessageEventType::Video:
+                    return QVariant::fromValue(e->content()->originalJson);
                 default:
                     return e->plainBody();
-                    break;
             }
         }
-        if( event->type() == EventType::RoomMember )
+    }
+
+    if( role == HighlightRole )
+        //return m_currentRoom->isEventHighlighted(event);
+
+    if( role == ReadMarkerRole )
+        return event->id() == lastReadEventId;
+
+    if( role == SpecialMarksRole )
         {
-            RoomMemberEvent* e = static_cast<RoomMemberEvent*>(event);
-            switch( e->membership() )
-            {
-                case MembershipType::Join:
-                    return QString("%1 (%2) joined the room").arg(e->displayName(), e->userId());
-                case MembershipType::Leave:
-                    return QString("%1 (%2) left the room").arg(e->displayName(), e->userId());
-                case MembershipType::Ban:
-                    return QString("%1 (%2) was banned from the room").arg(e->displayName(), e->userId());
-                case MembershipType::Invite:
-                    return QString("%1 (%2) was invited to the room").arg(e->displayName(), e->userId());
-                case MembershipType::Knock:
-                    return QString("%1 (%2) knocked").arg(e->displayName(), e->userId());
-            }
+        if (event->isStateEvent() &&
+                static_cast<const StateEventBase*>(event)->repeatsState())
+            return "hidden";
+        return event->isRedacted() ? "redacted" : "";
         }
-        if( event->type() == EventType::RoomAliases )
+
+    if( role == EventIdRole )
+        return event->id();
+
+    if( role == LongOperationRole )
+    {
+        if (event->type() == EventType::RoomMessage &&
+                static_cast<const RoomMessageEvent*>(event)->hasFileContent())
         {
-            RoomAliasesEvent* e = static_cast<RoomAliasesEvent*>(event);
-            return QString("Current aliases: %1").arg(e->aliases().join(", "));
+            auto info = m_currentRoom->fileTransferInfo(event->id());
+            return QVariant::fromValue(info);
         }
-        if( event->type() == EventType::Unknown )
+    }
+
+    if (role == AvatarRole)
+    {
+        if( event->type() == EventType::RoomMessage )
+        {
+            RoomMessageEvent* e = static_cast<RoomMessageEvent*>(event);
+            for (int i = 0; i < m_currentRoom->users().size(); i++){
+                 if (e->senderId() ==  m_currentRoom->users()[i]->id()){
+                     auto mediaid = m_currentRoom->users()[i]->avatarMediaId();
+                     if (!mediaid.isEmpty()) {
+                        auto url = QUrl("image://mtx/" + mediaid);
+                        return url;
+                     }
+                 }
+            }
```

Comment:
JFYI: an even fresher version of MessageEventModel just pushes the whole User object into AuthorRole so you won't need a separate AvatarRole and you won't need to iterate through 10000 users of Matrix HQ when rendering each message (actually, you don't need to do it already, `m_currentRoom->connection()->user()` would do the trick for you).

Reply:
uhh nice will look into that ... thanks

Reply:
Done